compiler/src/iree/compiler/Codegen/LLVMGPU/Passes.cpp

```diff
@@ -486,6 +506,10 @@ static void addLowerToLLVMGPUPasses(OpPassManager &pm, bool useROCM) {
 
   pm.addNestedPass<func::FuncOp>(createLLVMGPUVectorLoweringPass());
 
+  // THIS NEEDS TO RUN BEFORE SCF ->CF ON
+  addLowerAndOptimzeAddressComputation(pm);
+  // THIS NEEDS TO RUN BEFORE SCF ->CF OFF
```

Comment:
Could this run before as well? Like before software pipelining when doing matmul code?

Reply:
That I don't know.
This lowering will turn affine into llvm stuff and also we need to have all the memrefs stuff (i.e., after bufferization).
So I don't know how much flexibility there is here and if the pipelining can work within these constraints.
You tell me :).

Where do you think we should move it?

Reply:
It will lower affine to arithmetic right? Not to LLVM dialect? lowering affine late in pipeline flow should be fine. The pipelining doesn't rely on affine to work. I would move it here to replace the licm pass:
https://github.com/openxla/iree/blob/ea0b28fe850a06123383cb1cf7671995362c06b1/compiler/src/iree/compiler/Codegen/LLVMGPU/Passes.cpp#L329

Reply:
> It will lower affine to arithmetic right? Not to LLVM dialect? 

Yes, silly me.